### PR TITLE
KT-26629 Inspection to replace `==` operator on Double.NaN with `equals` call

### PR DESCRIPTION
**Pull request by devbridie**
YouTrack: https://youtrack.jetbrains.com/issue/KT-26629

Replaces fragments like `5.0 == Double.NaN` with `5.0.equals(Double.NaN)`.

### CODE REVIEW

**Comment by ilya-g** [comment]
- Have you considered using `number.isNaN()` function as a replacement?
- Does the inspection work for `!=` operator too? I believe it would be inconsistent to implement it only for `==`.

**Comment by devbridie** [comment]
Indeed, using `isNaN` seems better. I have used this as a replacement and have added support for `!=`.

File: idea/resources/META-INF/plugin-common.xml [hunk 1]
```diff
@@ -2350,6 +2350,15 @@
                      language="kotlin"
     />
 
+    <localInspection implementationClass="org.jetbrains.kotlin.idea.inspections.ConvertEqualsDoubleNaNInspection"
+                     displayName="Convert '== Double.NaN' to 'equals' call"
```
Comment (mglukhikh) [inline idea/resources/META-INF/plugin-common.xml]: This name contradicts with description and inspection logic.

File: idea/resources/META-INF/plugin-common.xml [hunk 2]
```diff
@@ -2350,6 +2350,15 @@
...
+                     groupPath="Kotlin"
+                     groupName="Probable bugs"
```
Comment (mglukhikh) [inline idea/resources/META-INF/plugin-common.xml]: I think it would be good to explain in description why it's a probable bug.

**Comment by devbridie** [comment]
Small ping, as I didn't comment when I hade made the changes!

**Comment by mglukhikh** [comment]
Thank you, I think now it can be accepted.

**Pull request merged**